src/message-stream.ts

```diff
@@ -384,3 +382,104 @@ export class MessageStream extends PassThrough {
     }
   }
 }
+
+/**
```

Comment:
how painful would it be to pull this into its own file? I'd be tempted to have a `message-stream.ts` and a `polling-message-stream.js`.

Reply:
It would be super easy! SGTM